### ISSUE TITLE
Target Correct CPU architecture with ftl deploy

### ISSUE THREAD

**Issue by alec-brooks** [block/ftl#1845]
The behaviour of `ftl deploy` is currently to build FTL modules with the CPU architecture of the system that the FTL cli is running on. This is problematic when building on a device that has a different CPU architecture than the environment where the FTL cluster is running.

Being able to specify the CPU architecture to build for as an argument in `ftl deploy` and `ftl build` would likely remediate this.

Though it would be very cool if it was possible for the FTL CLI to ask the running FTL cluster what CPU architecture it has before building so that users don't need to remember to pass in the flag.

**Comment by alecthomas** [comment]
Unfortunately for the general case across languages this would be very difficult, or impossible, to achieve.

**Comment by alecthomas** [comment]
OTOH we should definitely reject deploys from a different architecture.

**Comment by alec-brooks** [comment]
Yeah, I see what you mean, it would be very hard to do generically. I think just blocking deploys from other architectures would be a big improvement as right now it is easy to break things accidentally.
